crates/miden-testing/tests/scripts/faucet.rs

```diff
@@ -138,9 +150,224 @@ fn faucet_contract_mint_fungible_asset_fails_exceeds_max_supply() -> anyhow::Res
+                push.{aux}
+                push.{tag}
+                push.{amount}
+                # => [amount, tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
+
+                call.::miden::contracts::faucets::basic_fungible::distribute
+                # => [note_idx, pad(15)]
+
+                # truncate the stack
+                dropw dropw dropw dropw
+            end
+            ",
+        note_type = note_type as u8,
+        recipient = recipient,
+        aux = aux,
+        tag = u32::from(tag),
+        note_execution_hint = Felt::from(note_execution_hint)
+    );
+
+    let tx_script = ScriptBuilder::default().compile_tx_script(tx_script_code)?;
+    let tx_context = mock_chain
+        .build_tx_context(faucet.id(), &[], &[])?
+        .tx_script(tx_script)
+        .build()?;
+
+    let executed_transaction = tx_context.execute_blocking()?;
+
+    prove_and_verify_transaction(executed_transaction.clone())?;
+
+    let fungible_asset: Asset = FungibleAsset::new(faucet.id(), amount.into())?.into();
+
+    let output_note = executed_transaction.output_notes().get_note(0).clone();
+
+    let assets = NoteAssets::new(vec![fungible_asset])?;
+    let id = NoteId::new(recipient, assets.commitment());
+
+    assert_eq!(output_note.id(), id);
+    assert_eq!(
+        output_note.metadata(),
+        &NoteMetadata::new(faucet.id(), NoteType::Private, tag, note_execution_hint, aux)?
+    );
+
+    Ok(())
+}
+
+// TEST FOR create_basic_fungible_faucet FUNCTION
+// ================================================================================================
+
+#[test]
+fn create_basic_fungible_faucet_works() -> anyhow::Result<()> {
+    // Test that create_basic_fungible_faucet creates a properly configured faucet account
+    let init_seed: [u8; 32] = [0; 32];
+    let pub_key = miden_objects::crypto::dsa::rpo_falcon512::PublicKey::new(Word::from([0u32; 4]));
+    let auth_scheme = AuthScheme::RpoFalcon512 { pub_key };
+    let token_symbol = TokenSymbol::try_from("TST")?;
+    let decimals = 10u8;
+    let max_supply = Felt::new(200);
+    let account_storage_mode = AccountStorageMode::Public;
+
+    let (faucet_account, _) = create_basic_fungible_faucet(
+        init_seed,
+        token_symbol,
+        decimals,
+        max_supply,
+        account_storage_mode,
+        auth_scheme,
+    )?;
+
+    // Verify the faucet account is properly created
+    assert_eq!(
+        faucet_account.account_type(),
+        miden_objects::account::AccountType::FungibleFaucet
+    );
+
+    // Verify the faucet can be extracted and has correct metadata
+    let faucet_component = BasicFungibleFaucet::try_from(faucet_account.clone())?;
+    assert_eq!(faucet_component.symbol(), token_symbol);
+    assert_eq!(faucet_component.decimals(), decimals);
+    assert_eq!(faucet_component.max_supply(), max_supply);
+
+    Ok(())
+}
+
+#[test]
+fn prove_new_faucet_execution_burn_fungible_asset_succeeds() -> anyhow::Result<()> {
```

Comment:
Iiuc, this test is similar to `prove_faucet_contract_burn_fungible_asset_succeeds` except that it mints the assets directly instead of adding a faucet that basically already contains assets. It would be nice to deduplicate the code, but it may also mean that we'd run proving for both tests, which is a bit undesirable as mentioned in another comment. So this is probably fine.